flintrock/ec2.py

```diff
@@ -1014,6 +1014,24 @@ def get_clusters(*, cluster_names: list=[], region: str, vpc_id: str) -> list:
     return clusters
 
 
+def validate_tags(ctx, param, value):
+    """
+    Validate and parse optional ec2 tags as supplied on command line.
+    """
+    ec2_tags = value
+    for ec2_tag in ec2_tags:
+        if ec2_tag.count(',') != 1 or not ec2_tag.split(',')[0]:
+            raise click.BadParameter(
+                "tags need to be specified as 'Key,Value' pairs "
+                "separated by a single comma. Key cannot be empty.")
+
+    ec2_tags = [{'Key': tag.split(',')[0].strip(),
+                 'Value': tag.split(',')[1].strip()}
+                for tag in ec2_tags]
```

Comment:
We won't catch keys that are whitespace: `--ec2-tag '   ,ihavenokeylulz'`

Is there a way to unify the logic so that we parse the tags only one time, instead of once for error checking and then again to create the data structure we need? That way we avoid these issues.

Perhaps something like

```python
key, value = [word.strip() for word in tag.split(',', maxsplit=1)]
```

followed by a) whatever error checking we need to do, and then b) using `key` and `value` to build the data structure we need. That way we parse only once.